home-manager/ssh.nix

```diff
@@ -71,22 +72,31 @@ in
 
     # https://www.clear-code.com/blog/2023/4/3/recommended-ssh-config.html
     # https://gitlab.com/clear-code/ssh.d/-/blob/main/global.conf?ref_type=heads
-    extraConfig = ''
-      PasswordAuthentication no
-
-      # default: "ask"
-      StrictHostKeyChecking yes
-
-      # https://serverfault.com/a/1109184/112217
-      CheckHostIP no
-
-      # `UseKeychain` only provided by darwin ssh agent, in Linux and pkgs.openssh, it isn't
-      IgnoreUnknown UseKeychain
-      UseKeychain yes
-    '';
+    extraConfig =
+      ''
+        PasswordAuthentication no
+
+        # default: "ask"
+        StrictHostKeyChecking yes
+
+        # https://serverfault.com/a/1109184/112217
+        CheckHostIP no
+      ''
+      + (
+        # `UseKeychain` only provided by darwin ssh agent, in Linux and pkgs.openssh, it isn't
+        lib.strings.optionalString pkgs.stdenv.hostPlatform.isDarwin ''
+          IgnoreUnknown UseKeychain
+          UseKeychain yes
+        ''
+      );
 
     # No problem to register the same *.pub in different services
     matchBlocks = {
+      # Workaround for GH-950
+      # See https://github.com/NixOS/nixops/issues/395#issuecomment-2632428059
+      "*".extraOptions.IgnoreUnknown =
+        "gssapikexalgorithms,gssapiauthentication,gssapidelegatecredentials";
```

Comment:
* Unnecessary in darwin, but adding for both with my laziness
* I often afraid the order of matchBlocks, however don't consider for now.
* I think I can merge following sharedConfig blocks for each host. However ignored for now.